### DISCUSSION TITLE
Random words recognized while not speaking or speaking non-English

### DISCUSSION THREAD

**Discussion by rudymohammadbali** [openai/whisper#1455] [Q&A]
I am facing an issue with the transcription of audio using the Whisper module. It randomly recognizes words like "Thanks you" and "Thanks for watching" even when I'm not speaking or speaking in a language other than English.

Has anyone else encountered this problem? If so, I would greatly appreciate any guidance or suggestions on how to resolve this issue.

I have tried adjusting the settings and parameters of the Whisper module, but I haven't been able to find a solution. It seems to be an issue with the speech recognition algorithm incorrectly interpreting non-speech sounds or capturing random words.

Is there a way to fine-tune the module to improve its accuracy and avoid recognizing irrelevant words? Or perhaps there are specific settings or techniques I should consider to mitigate this issue?

Any help or insight into this problem would be highly appreciated. Thank you in advance for your assistance!

# Here is a example of what it recognized while not speaking or speaking in other language:

Nice, DJ.    Thank you.  AHHH! AHHH!  What? Thank you. Thank you. Thank you. Thank you.  a lot of the function if you're fancy to any The better my trip will get. Thank you. Thank you. I know, probably. Put it in. Put it in. Thank you.  Thanks for watching!  I'll see you later.   I'll see you next time. Thanks for watching! Thanks for watching!     thank you  Thank you. I'll see you next time. Thank you. Thank you. Thanks for watching!     Thank you. Now Quyyy Comment and Subscribe A little bit of history, right? so good it's funny English Story. Write me. I get you. Discussion. For the health of wise people. But look, here's a problem. Here's a problem. Take me right down the road. But at night. While not speaking. Description. Description of the Lord. Ava. Almighty.  Let's just spend the eveningbreak Mincey cantat Aw All right What is it? Universal. I think it's 10 now. I'm not allowed. I'm shit. One, one. I'm not allowed. I'm not allowed. It is a Need for Humanity. I'm not sure how I'll be able to do it. I don't know if I'll be able to do it. I don't know if I'll be able to do it. I don't know. I don't know. I'm not sure if I can do it. I'm not sure if I can do it. Welcome! Hey! I can do it with Chef D, too. You can do it with Chef Sun. Ah. Thank you. I'm not sure what to say. I'm not sure. Do you have any questions? I don't have any. I don't have any. I don't have any. I don't have any. For real... Four hands since from there I'll see you next time.  But it's not like there's thattheme I want two big keys! So it all there Thanks for watchinglaughs We brought Ronald, Is this the first time? Yes, the first time. I think it's the first time. I'm not sure if it's the first time. so se files are fully deployed? Yeah, they are pre baking sheets and they still weren't as That's what it feels like. Thank you. I'll see you next time. Thank you. Thank you. I'll see you next time. I'll see you next time. Thank you. Please see the complete disclaimer at PissedConsumer.com I'll see you next time. I'll see you next time. I'll see you next time. I'll see you guys next time. I'll see you guys next time. I'll take a look. Do you want a bit? Is that cool? Yeah. Thank you. Okay, excuse me. You're welcome. Okay. I'll go and look for you. I'll go and look for you. Thank you. We'll just let him go. Thank you. Thank you. Richard Peaceful… Thank you. Maybe… Thank you. I don't. Good stuff.     I'll see you next time. Thanks for watching! Thank you for watching!

**Comment by phineas-pta** [comment]
it's the hallucinations problem, happened when there's silence / non speech segments

see #679

**Reply by rudymohammadbali** [reply]
I did not find a solution. Please tell me the answer if you know it and thanks.

**Comment by glangford** [comment]
see also
* https://github.com/openai/whisper/discussions/928

**Comment by rudymohammadbali** [comment]
If no fix from OpenAI so I can do nothing but else wait for a solution.

**Reply by ryanheise** [reply]
Waiting for OpenAI is not the only option, as many people have shared solutions to the hallucination problem on this discussion board. Now that you know what the problem is called (thanks to the people who answered you), you can try searching for that keyword here.

**Reply by rudymohammadbali** [reply]
I'm making real-time speech recognition. so the recorded audio file is about 2-3 seconds. and I don't know how to process this audio file to remove the silent parts.

**Comment by misutoneko** [comment]
You might want to take a look here:
https://github.com/openai/whisper/discussions/1488

So basically it seems like a bug somewhere in how non_speech_tokens is handled.
I haven't received much feedback on that btw, so all I can say about it is that it works for me.
My samples are also very short, so maybe it helps you too.

**Reply by rudymohammadbali** [reply]
thanks for answer.

**Comment by MiguelQueiroz** [comment]
yeah, I use a voice assistant that uses openAI  whisper for speech detection/commands, and when there is silence I get a response back talking about Review No.106161 on PissedConsumer.com or other type of review but it mostly goes to PissedConsumer.com most of the time dont know why! This is because the voice transcription was detected as such! 😩

**Comment by Mohdnihal03** [comment]
Did anyone found the solution to avoid "Thanks for watching"?

**Reply by misutoneko** [reply]
Yes I'd say so. Perhaps it isn't a universal fix, but I found a solution that works for _me_ back in... let's see, July 2023, so over two years ago :D
(See my previous post in this thread.)

Now that method _does_ come with its own set of limitations (for example, it doesn't work with turbo or v3).
I just happen to like it better than the alternatives (like using an aggressive VAD strategy, adjusting compress ratio or no_speech thresholds, or even switching over to parakeet...)
